File: 2026/TUF/JavaScript/Array/problem_02.js

//find the second largest Number

// let arr=[1,6,5];  edge cases for the second largest number 
let arr=[1,2,3,4,5,6];

let max=arr[0];  //6
let min=0; //1

for(let i=1;i<arr.length;i++){
    if(max<arr[i]){
        min=max;
        max=arr[i];
    }else if(arr[i]>min && arr[i]!=max) {
        min=arr[i]
    }
}
console.log(min)
console.log(max)

File: 2026/TUF/Java/Array/Problem_02.java

// Online Java Compiler
// Use this editor to write, compile and run your Java code online

class Problem_02 {
    
    static int secondLargest(int[] arr){
        
        int max=arr[0];  //6
int min=0; //
        for(int i=1;i<arr.length;i++){
    if(max<arr[i]){
        min=max;
        max=arr[i];
    }else if(arr[i]>min && arr[i]!=max) {
        min=arr[i];
    }
    
}
return min;
    }
    
    public static void main(String[] args) {
        System.out.println("Try programiz.pro");
        int[] arr={1,6,5};
        System.out.println("Second Largest:"+secondLargest(arr));
    }
}